### DISCUSSION TITLE
Any way getting this working on a M1/M2 Mac?

### DISCUSSION THREAD

**Discussion by elchupacabrinski** [adieyal/sd-dynamic-prompts#305] [Q&A]
I've been using Dynamic Prompts on Linux for quite a while without a Problem, however on Mac OS it doesn't seem to work due to lack of CUDA support

I'm always getting a long error message ending with: "AssertionError: Torch not compiled with CUDA enabled" Is there a build of this that works on Apple Silicon or a setting I need to enable?

Edit: It throws this error only when prompt magic is enabled!

**Comment by adieyal** [comment]
I've  created a branch which  I hope fixes this. Could you please test and let me know?
To test the branch, do the following

```
cd extensions/sd-dynamic-prompts
git pull -a
git checkout origin/bug/device
```

then restart webui

to return to the main branch


`git checkout main`

**Comment by pto2k** [comment]
Hi, 

I am currently using Webui DirectML on Windows with an AMD graphics card. I encountered the same error while using the Magic Prompt option: "AssertionError: Torch not compiled with CUDA enabled". (If I disable the 'Magic prompt' and enable the 'I'm feeling lucky' option, that works.)

Although I attempted to fix this problem by trying out the bug branch, it did not work for me. 

I also tried to use another script from https://github.com/Spaceginner/MagicPrompt-awebui/, which worked fine for me.

I hope this information is helpful. Thank you for your hard work!

**Reply by adieyal** [reply]
Thanks - looks like that script is using another library to load Magic Prompt. I'll take a closer look once I get my hands on a non-Nvidia machine. I've opened an issue here: https://github.com/adieyal/sd-dynamic-prompts/issues/329

**Comment by akx** [comment]
Should work with https://github.com/adieyal/sd-dynamic-prompts/issues/329.
